### DISCUSSION TITLE
Print weights summary

### DISCUSSION THREAD

**Discussion by thesimonho** [Lightning-AI/pytorch-lightning#5786] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Weights summary gets printed when Trainer calls `fit()`, but the output is not persistent as I have everything wrapped up in ray tune, which overwrites the contents of the output cell in jupyter

Is there something we can call to manually print the weights summary of a particular model without having to fit the model every time?

**Comment by rohitgr7** [comment, marked as answer]
just do 
```python
model = YourLightningModule(...)
model.summarize(mode=...)
```
`mode='top', 'full'`
